### PR TITLE
Added custom buttons to VM details page

### PR DESCRIPTION
**Pull request by chalettu**
PV https://www.pivotaltracker.com/story/show/142774757

### CODE REVIEW

**Comment by chalettu** [comment]
@miq-bot add_label fine/no
@miq-bot add_label my services / details
@miq-bot add_label enhancement

**Comment by chalettu** [comment]
custom_buttons
Screen shot of an example of custom buttons on VM Details page

**Comment by chriskacerguis** [comment]
@ManageIQ/core-ux for your review.

**Comment by serenamarie125** [comment]
@chalettu the mock is showing custom button and button groups without icons, which looks great!  Does the code support displaying an icon if there is one as well?

@chriskacerguis do we have a separate issue or PV for adding custom buttons for VMs on the service details page?

File: client/app/services/vms.service.js [hunk 1]
```diff
@@ -46,6 +46,8 @@ export function VmsService(CollectionsApi, RBAC) {
         'compliances',
         'cpu_affinity',
         'created_on',
+        'custom_actions',
```
Comment (AllenBW) [inline client/app/services/vms.service.js]: I belieeeeeeve adding these new attributes will break mock api get requests... probably should update that 😬

Comment (chalettu) [inline client/app/services/vms.service.js]: Yeah, they will.  I have an update for mock api that I will submit that should fix these

Comment (AllenBW) [inline client/app/services/vms.service.js]: 👍

**Comment by Loicavenel** [comment]
@serenamarie125 can you approve it quickly, thanks

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit 223c3c1425aa5c4945005764fa4349cba08742fc
Author: Chris Kacerguis <chriskacerguis@users.noreply.github.com>
Date:   Fri May 12 13:56:11 2017 -0500

    Merge pull request #750 from chalettu/vm-custom-buttons
    
    Added custom buttons to VM details page
    (cherry picked from commit 4918c1ac4444398983b63540adebb9c03bc94ad9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1450502
```

**Comment by simaishi** [comment]
@chalettu @chriskacerguis please take a look at Travis failure for `fine` branch: https://travis-ci.org/ManageIQ/manageiq-ui-service/jobs/231694009.

```
ERROR in ./client/app/services/vm-details/vm-details.component.js
/home/travis/build/ManageIQ/manageiq-ui-service/client/app/services/vm-details/vm-details.component.js
  102:12  error  'lodash' is not defined  no-undef
```

**Comment by chalettu** [comment]
Not sure what do on this one @chriskacerguis.  I double checked the code and we are defining lodash.  This seems like something going wrong with Travis.

**Comment by chriskacerguis** [comment]
I restarted the job...let's see what we see

**Comment by simaishi** [comment]
`yarn` command failed trying to compile SUI, so I don't think it's Travis... `master` branch has more change in that file which doesn't exist in `fine`. Does this PR depends on something that's not in Fine branch?

**Comment by chriskacerguis** [comment]
No.  Basically, it's saying it can't find lodash but, it is installed, and injected...so not sure the issue.

**Comment by chalettu** [comment]
Figured it out.  Its missing one small thing in that file that probably due to some other changes for vm details not being in fine.   @simaishi how do you want me to go about fixing it?

**Comment by simaishi** [comment]
Can you please create a Fine PR, referencing this?

**Comment by simaishi** [comment]
Depends on: https://github.com/ManageIQ/manageiq/pull/14817

**Pull request merged**